internal/app/frontend/frontend_service_test.go

```diff
@@ -88,6 +89,97 @@ func TestDoCreateTickets(t *testing.T) {
 	}
 }
 
+func TestCreateBackfill(t *testing.T) {
+	cfg := viper.New()
+	store, closer := statestoreTesting.NewStoreServiceForTesting(t, cfg)
+	defer closer()
+	ctx := utilTesting.NewContext(t)
+	fs := frontendService{cfg, store}
+
+	var testCases = []struct {
+		description     string
+		request         *pb.CreateBackfillRequest
+		result          *pb.Backfill
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description:     "nil request check",
+			request:         nil,
+			expectedCode:    codes.InvalidArgument,
+			expectedMessage: "request is nil",
+		},
+		{
+			description:     "nil backfill - error is returned",
+			request:         &pb.CreateBackfillRequest{Backfill: nil},
+			expectedCode:    codes.InvalidArgument,
+			expectedMessage: ".backfill is required",
+		},
+		{
+			description:     "createTime should not exist in input",
+			request:         &pb.CreateBackfillRequest{Backfill: nil},
+			expectedCode:    codes.InvalidArgument,
+			expectedMessage: ".backfill is required",
+		},
+		{
+			description:     "createTime should not exist in input",
+			request:         &pb.CreateBackfillRequest{Backfill: &pb.Backfill{CreateTime: ptypes.TimestampNow()}},
+			expectedCode:    codes.InvalidArgument,
+			expectedMessage: "backfills cannot be created with create time set",
+		},
+		{
+			description:     "empty Backfill, no errors",
+			request:         &pb.CreateBackfillRequest{Backfill: &pb.Backfill{}},
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description: "normal backfill",
+			request: &pb.CreateBackfillRequest{
+				Backfill: &pb.Backfill{
+					SearchFields: &pb.SearchFields{
+						StringArgs: map[string]string{
+							"search": "me",
+						}}}},
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.description, func(t *testing.T) {
+			res, err := fs.CreateBackfill(ctx, tc.request)
+			if tc.expectedCode == codes.OK {
```

Comment:
Avoid clauses like this in tests: instead this should be two separate test cases.